PQAnalysis/io/traj_file/trajectory_reader.py

```diff
@@ -198,38 +199,45 @@ def frame_generator(
 
         last_cell = None
 
-        for filename in self.filenames:
+        for i, filename in enumerate(self.filenames):
+
+            if trajectory_start >= frame_index + number_of_frames[i]:
+                frame_index += number_of_frames[i]
+                continue
+
+            if trajectory_stop <= frame_index:
+                break
 
             # Read the file again to get the frames
             with open(filename, "r", encoding="utf-8") as self.file:
                 frame_lines = []
 
                 progress_bar = tqdm(
-                    total=self.length_of_traj,
+                    total=trajectory_stop - trajectory_start,
                     disable=not self.with_progress_bar
                 )
```

Comment:
I think this progress bar is broken for multiple files (and was always broken).

So actually, the progress_bar is initialized inside the `for i, filename in ...`, which means that it will be re-initialized for every file, which will be read. But then we should not use `self.length_of_traj` or as it is implemented now `trajectory_stop - trajectory_start`, but we would need to use somehow the list of number of frames, with a start and stop criterium.

Reply:
Oh, now I see it. Wouldn't it make more sense if it is the progress over the whole trajectory and not only a specific file?

Reply:
@97gamjak I will initialize it outside of the loop for now.

Reply:
> Oh, now I see it. Wouldn't it make more sense if it is the progress over the whole trajectory and not only a specific file?

Yeah I think there isn't one best solution. I think both approaches have there advantages and disadvantages. But I think putting it outside of the loop will make it easier.